docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
+   - 1c. User adds task without start and end time  
+Time is set to empty  
+   - 1d. User inputs wrong details format  
+PLANus shows invalid command message  
+
+
+### Use case: List task
+
+Main success scenario
+
+1. User requests to list tasks
+2. PLANus shows tasks
+
+Use case ends.
+
+Extensions
+   - 2a. List is empty - PLANus displays empty list
+
+
+### Use case: Edit task
+
+Main success scenario
+
+1. User inputs the task index, and what user wants to change
+2. PLANus changes the task details
+3. PLANus shows the task edited message. 
+
+Use case ends.
+
+Extensions
+   - 1a. User inputs an invalid index
+Shows invalid command message
+   - 1b. User inputs wrong details format
+Shows invalid command message
+
+
+### Use case: Help
+
+Main success scenario
+
+1. User inputs help command
+2. PLANus displays help message
+
+Use case ends.
+
+
+### Use case: Search
+
+Main success scenario
+
+1. User inputs search command with keyword
+2. PLANus searches the task list for keyword
+3. PLANus displays results
+
+Use case ends.
+
+Extensions
+   - 2a. PLANus is unable to find any matches
+Shows no tasks found message
+
+
+### Use case: Delete
+
+Main success scenario
+
+1. User inputs delete command with index
+2. PLANus deletes the task
+3. PLANus shows task deleted message
+
+Use case ends.
+
+Extensions 
+   - 1a. User inputs invalid index
+PLANus shows invalid task number message
+
+
+### Use case: Clear 
+
+Main success scenario
+
+1. User inputs clear command
+2. PLANus clears all tasks
+3. PLANus shows tasks cleared message
+
+Use case ends. 
+
+
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+    - The system should respond within two seconds.
+    - The system should be easy to use, do not require much effort to learn.
+    - Features should be able to be implemented before project deadline.
 
 ## Glossary
 
-* *glossary item* - Definition
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### Launch and Shutdown
+
+1. Initial launch
+	
+    - Download the jar file and copy into an empty folder. Type `java -jar planus.jar`
+    int the command line window and press enter, welcome message will be displayed.
+
+2. Shutdown
+
+    - Type “bye” into the command line and the program should exit.
+
+### Adding a task
+
+1. Add a task into the list by using add command using only description
+
+    - Test case: add homework  
+    Expected: homework task should be added to the list with default date being set 
+    to the current date.
+
+2. Add a task into the lists using add command with description and date.
+
+    - Test case: add homework d/25-12-2020  
+    Expected: homework task should be added to the list with the date set as 25-12-2020.
+
+    - Test case: add homework d/25-13-2020  
+    Expected: Since there is no 13th month, the task is not added to the list, 
+    and an error message is shown.
+
```

Comment:
It might be helpful to give a brief description of what the identifier (eg: `/d`) means. For example, can give an explanation that `/d` represents the date.